packages/contracts-diamond/contracts/assets/ERC721/ERC721DropLib.sol

```diff
@@ -0,0 +1,118 @@
+// Store address + maxCuserClaim
+// prove data
+// update on-chain tracking
+
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import {MerkleProof} from "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+
+library ERC721DropLib {
+    /// @custom:storage-location erc7201:erc721.drop.storage
+    struct DropStorage {
+        bytes32 merkleRoot;
+        mapping(address => uint256) maxUserClaim;
```

Comment:
You do not to store this. Should be part of the struct stored in the Merkle tree (address, uint256)